guides/v2.2/coding-standards/technical-guidelines.md

```diff
@@ -715,6 +715,424 @@ You need to read configuration from different sources (like database or filesyst
+    <see selector="{{StorefrontProductInfoMainSection.price}}" userInput="{{ClamberWatch.price}}" stepKey="seeProductPrice" />
+    <waitForElementNotVisible selector="{{StorefrontProductMediaSection.gallerySpinner}}" stepKey="waitGallerySpinnerDisappear" />
+    <seeElement selector="{{StorefrontProductMediaSection.gallery}}" stepKey="seeProductGallery" />
+    <seeElement selector="{{StorefrontProductMediaSection.productImage(ClamberWatch.productImage)}}" stepKey="seeProductImage" />
+    <click selector="{{StorefrontProductMediaSection.productImage(ClamberWatch.productImage)}}" stepKey="openFullscreenImage" />
+    <waitForPageLoad stepKey="waitForGalleryLoaded" />
+    <seeElement selector="{{StorefrontProductMediaSection.productImageFullscreen(ClamberWatch.productImage)}}" stepKey="seeFullscreenProductImage" />
+</test>
+```
+
+{% endcollapsible %}
+
+11.3.9.12. `stepKey` names MUST be descriptive. This helps with readability and clarity.
+
+{% collapsible Examples: %}
+
+### Recommended
+
+```xml
+<click selector="{{StorefrontNavigationSection.topCategory(SimpleSubCategory.name)}}" stepKey="clickSimpleSubCategoryLink" />
+<waitForPageLoad stepKey="waitForSimpleSubCategoryPageLoad" />
+<click selector="{{StorefrontCategoryMainSection.productLinkByHref(SimpleProduct.urlKey)}}" stepKey="clickSimpleProductLink" />
+<waitForPageLoad stepKey="waitForSimpleProductPageLoad" />
+
+<!-- Perform some actions / Assert product page -->
+
+<click selector="{{StorefrontNavigationSection.topCategory(CustomCategory.name)}}" stepKey="clickCustomCategoryLink" />
+<waitForPageLoad stepKey="waitForCustomCategoryPageLoad" />
+<click selector="{{StorefrontCategoryMainSection.productLinkByHref(CustomSimpleProduct.urlKey)}}" stepKey="clickCustomSimpleProductLink" />
+<waitForPageLoad stepKey="waitForCustomSimpleProductPageLoad" />
+```
+
+### Not recommended
+
+```xml
+<click selector="{{StorefrontNavigationSection.topCategory(SimpleSubCategory.name)}}" stepKey="clickCategoryLink1" />
+<waitForPageLoad stepKey="waitForPageLoad1" />
+<click selector="{{StorefrontCategoryMainSection.productLinkByHref(SimpleProduct.urlKey)}}" stepKey="clickProductLink1" />
+<waitForPageLoad stepKey="waitForPageLoad2" />
+
+<!-- Perform some actions / Assert product page -->
+
+<click selector="{{StorefrontNavigationSection.topCategory(CustomCategory.name)}}" stepKey="clickCategoryLink2" />
+<waitForPageLoad stepKey="waitForPageLoad3" />
+<click selector="{{StorefrontCategoryMainSection.productLinkByHref(CustomSimpleProduct.urlKey)}}" stepKey="clickProductLink2" />
+<waitForPageLoad stepKey="waitForPageLoad4" />
+```
+
+{% endcollapsible %}
+
+#### 11.3.10. Extending
+
+11.3.10.1. You SHOULD use extends in your new test or action group when at least one of the following conditions is applicable to your case:
+
+* You want to keep the original test without any modifications.
```

Comment:
This is not entirely true. Sometimes we remove steps, don't we?

Reply:
This is saying that `extends` leaves the original test unchanged. I took it from our devdocs.

What do you want it to say instead?

Reply:
Never mind. I misunderstand you. Nothing needs to be changed here.
Now i read the entire paragraph, what you are saying is "keep everything unchanged" is ONE of the case, which is correct. There are other cases that you can make modifications with extends.